2016/notes-on-testing-with-jest.md

```diff
@@ -0,0 +1,280 @@
+# Notes on Jest
+
+During the past 4 months, I have been working on the Flipkart desktop website team. For me it meant finally working on code at scale. If there was any place where TDD would be appreciated, it was here, on user facing critical code. The tests were being written on a setup of Karma, Mocha, Sinon and Enzyme. It worked pretty great, but the major peeve for all of us, was that, it was so slow!
+
+The team had already tried using Jest before, had faced many issues with the automocking and given up. But then Jest made a turnaround and with a great community effort released improved and [new versions](https://facebook.github.io/jest/blog/2016/10/03/jest-16.html). From all the blogposts, we found that Jest's main focus was testing performance and something new called ["Snapshot testing"](https://facebook.github.io/jest/blog/2016/07/27/jest-14.html). Also, mainly it promised to be "Painless JavaScript Testing" with a special focus on React.
+
+This post wants to be an FAQ guide which would have given us extra help during this transition.
+
+## FAQs
+
+The FAQs are categorised by the facet of testing that they cover. These may also apply to any testing in general.
+
+### Fundamentally speaking
+
+__Why should we move to `Jest` when we already have a working setup?__
+
+A move to Jest would mean a somewhat change in paradigm and would involve team effort. We justified the effort with the following reasons
+
+  - Jest is way faster than our current setup. It runs test suites in a parallel and the community is focused on improving developer experience through performance.
+  - I really liked the interactive CLI that comes with it. In `watch` mode, Jest runs only the tests associated with your code changes. But it allows you to interact with the tests by providing functionality to update snapshots, search for a particular suite, run all tests if you want to.
+  - [Great documentation](https://facebook.github.io/jest/docs/getting-started.html)
+  - Snapshot testing for components. It is kind of tailor made for React apps.
+  - Easy configuration and great defaults. Running `jest` alone would be enough to run test files. Mocking is not a marriage of complexity and confusion.
```

Comment:
Yep, coverage is awesome. Will include a bit about that. 
"Zero configuration" is true actually!